resources/views/components/truncate-dynamic.blade.php

```diff
@@ -2,12 +2,24 @@
     x-data="{
         value: '{{ $slot }}',
         init() {
-            new ResizeObserver(() => this.truncate()).observe(this.$el);
+            new ResizeObserver(() => this.throttledTruncate()).observe(this.$el);
 
-            document.addEventListener('resize', () => this.truncate());
+            window.addEventListener('resize', () => this.throttledTruncate());
 
             this.truncate();
         },
+        throttleTimeout: null,
+        throttledTruncate() {
+            if (this.throttleTimeout !== null) {
+                clearTimeout(this.throttleTimeout);
+            }
+
+            this.throttleTimeout = setTimeout(() => {
+                this.truncate();
+
+                this.throttleTimeout = null;
```

Comment:
This causes an infinite number of timeouts to start running :thinking: should this method not also have `clearTimeout(this.throttleTimeout);`? I'm not even sure if setting it to null is necessary, as the next resize would clear the previous timeout and start a new one :thinking: :thinking: 

if you want to recreate how I found this, open a browser window, show go to a page with truncating, dev tools, mobile, expand and resize the window and it will begin to hang. See video (i did resize more after it froze but the video capture didn't record it - basically it hangs): https://www.loom.com/share/c16fcce1854e43288514de177f0ad631